### ISSUE TITLE
[Feature Request] Implement `hydra.utils.get_object` to locate arbitrary object.

### ISSUE THREAD

**Issue by Jasha10** [facebookresearch/hydra#1975]
The [`hydra.utils.get_class`](https://github.com/facebookresearch/hydra/blob/abb36af33e83f391a22d6269fd2f317378171ec9/hydra/utils.py#L21) and [`hydra.utils.get_method`](https://github.com/facebookresearch/hydra/blob/abb36af33e83f391a22d6269fd2f317378171ec9/hydra/utils.py#L32) functions are useful to look up python objects based on dot-paths.
```python
>>> import hydra
>>> hydra.utils.get_method("numpy.float32")
<class 'numpy.float32'>
```
Internally, these functions use the same mechanism as the [`hydra.utils.instantiate`](https://hydra.cc/docs/advanced/instantiate_objects/overview/) function.

These two functions fail when looking up an object that is not a type and is not callable:
```python
>>> hydra.utils.get_method("tensorflow.float16")
...
ValueError: Invalid type (<class 'tensorflow.python.framework.dtypes.DType'>) found for tensorflow.float16
```

I propose a `hydra.utils.get_object` function that has the same behavior as `get_class` and `get_method` but is more permissive with regard to non-callable objects.

For a motivating use-case, see [this stackoverflow post](https://stackoverflow.com/a/70763884/4256346).

**Comment by Zekrom-7780** [comment]
@Jasha10 , please can I work on this issue?

**Comment by Jasha10** [comment]
@Zekrom-7780 thank you for volunteering -- this feature has already been implemented (PR #2520).
I will close this issue now.
